### PR TITLE
[Secrets] Doc BugBash Fixes

### LINKED ISSUE #16495: [Doc][azsecrets] outdated code sample for Update Secret metadata
- What happened?
copy and paste code snippet from Update Secret metadata sample but it doesn't build

- What did you expect or want to happen?
correct code:
```diff
	properties := azsecrets.Properties{
		ContentType: &contentType,
		SecretAttributes: &azsecrets.Attributes{
			Enabled: to.BoolPtr(false),
		},
	}
	updateResp, err := client.UpdateSecretProperties(context.Background(), "secretName", properties, nil)
```

logging sample is also incorrect
image

### PR DESCRIPTION
**Pull request by seankane-msft**
Closes #16495 and #16492
- [ ] Consistently refers to the Azure Identity package as `azidentity`
- [ ] Fixes bugs in sample code for `UpdateSecretProperties` and Logging example

### CODE REVIEW

File: sdk/keyvault/azsecrets/README.md [hunk 1]
```diff
@@ -164,16 +164,17 @@ contentType := "text/plain"
 
 // We will also disable the secret for further use
 
-properties := azsecrets.SecretProperties{
-    ContentType: &contentType,
-    SecretAttributes: &azsecrets.SecretAttributes{
-        Attributes: &azsecrets.Attributes{
-            Enabled: to.BoolPtr(false),
-        },
+properties := Properties{
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azsecrets/README.md:167]: Isn't the package name still required here and below?

File: sdk/keyvault/azsecrets/README.md [hunk 2]
```diff
@@ -236,12 +237,12 @@ To obtain more detailed logging, including request/response bodies and header va
 ```go
 import azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
```
Comment (jeremymeng) [inline sdk/keyvault/azsecrets/README.md:238]: `azlog` here, but `log` below

Comment (seankane-msft) [reply sdk/keyvault/azsecrets/README.md:238]: Thanks, good catch!

File: sdk/keyvault/azsecrets/models.go [hunk 3]
```diff
@@ -88,9 +88,13 @@ type Attributes struct {
 }
 
 func (s Attributes) toGenerated() *internal.SecretAttributes {
+	var recLevel *internal.DeletionRecoveryLevel
+	if s.RecoveryLevel != nil {
+		recLevel = s.RecoveryLevel.toGenerated().ToPtr()
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azsecrets/models.go:93]: Not saying you have to do this, but you could simplify this a bit by having `toGenerated()` take its receiver by pointer and perform the `nil` check there (it's legal to call a method on a `nil` thing as long as you pass by pointer and check for `nil`).  Then you just do `RecoveryLevel: s.RecoveryLevel.toGenerated()`.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**